decimal_msgpack.go

```diff
@@ -0,0 +1,92 @@
+package decimal
+
+import (
+	"errors"
+)
+
+var (
+	errShortBytes = errors.New("msgp: too few bytes left to read object")
+)
+
+// MarshalMsg implements msgp.Marshaler
+func (d Decimal) MarshalMsg(b []byte) (o []byte, err error) {
+	o = require(b, d.Msgsize())
+	str := d.String()
+	sz := len(str)
+	// limit to 30 digits
+	// note, if d.IntPart size large than 30, will be lose.
+	if sz > 30 {
+		sz = 30
+		// if last char is '.' then limit to 20 digits
+		if str[29] == '.' {
+			sz = 29
+		}
+
+		str = str[:sz]
+	}
+
+	o, n := ensure(b, 1+sz)
+	o[n] = byte(0xa0 | sz)
+	n++
+
+	return o[:n+copy(o[n:], str)], nil
+}
+
+// UnmarshalMsg implements msgp.Unmarshaler
+func (d *Decimal) UnmarshalMsg(b []byte) (o []byte, err error) {
```

Comment:
Could you stick to one type of return value (names/unnamed)? Across the entire library, we use an unnamed one, so I personally opt for this approach.

Reply:
already Change MsgPack MarshalMsg and UnmarshalMsg function's return value to unnamed.